lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5530,6 +5530,22 @@ void ConformanceChecker::resolveValueWitnesses() {
               .withEnterIsolation(*enteringIsolation));
       }
 
+      // FIXME: almost right...
+      // if (requirement->isDistributed()) {
+      //   // When witnessing a 'distributed' requirement, we also immediately
+      //   // register the witness for the distributed_thunk. Doing is more correct,
+      //   // as otherwise we might end up trying to modify an already completed
+      //   // conformance once we get to visit the thunk itself.
+      //   if (auto func = dyn_cast<AbstractFunctionDecl>(requirement)) {
+      //     auto thunkRequirement = func->getDistributedThunk();
+      //     assert(thunkRequirement->isDistributedThunk());
+      //     auto distributedWitness = dyn_cast<AbstractFunctionDecl>(witness);
+      //     auto thunkWitness = Witness(distributedWitness->getDistributedThunk());
+      //     assert(dyn_cast<FuncDecl>(thunkWitness.getDecl())->isDistributedThunk());
+      //     Conformance->setWitness(thunkRequirement, thunkWitness);
+      //   }
+      // }
```

Comment:
Minor cleanup TODO here, and then unblock `// TODO(distributed): properly handle isComplete() for distributed`